kombu/compat.py

```diff
@@ -102,8 +100,8 @@ def __init__(self, connection, queue=None, exchange=None,
 
         self.queue = queue or self.queue
         self.exchange = exchange or self.exchange
-        self.exchange_type = exchange_type or self.exchange_type
-        self.routing_key = routing_key or self.routing_key
+        self.exchange_type = exchange_type or 'direct'
```

Comment:
`self.exchange_type` was defined twice, I pushed an update to remove both. I think the proposed change is easier to read, because the default value is on the same line. If I'm pointed to a suitable place, I can add a test.